src/Microsoft.AspNet.Mvc.Core/ActionResults/PartialViewResult.cs

```diff
@@ -50,10 +51,22 @@ public class PartialViewResult : ActionResult
             var viewEngine = ViewEngine ??
                              context.HttpContext.RequestServices.GetRequiredService<ICompositeViewEngine>();
 
+            var logger = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>()
+                .CreateLogger<PartialViewResult>();
+
             var viewName = ViewName ?? context.ActionDescriptor.Name;
-            var view = viewEngine.FindPartialView(context, viewName)
-                                 .EnsureSuccessful()
-                                 .View;
+            var viewEngineResult = viewEngine.FindPartialView(context, viewName);
+            if (!viewEngineResult.Success)
+            {
+                logger.LogError(
+                    "The partial view '{PartialViewName}' was not found. Searched locations: {SearchedViewLocations}",
+                    viewName,
+                    viewEngineResult.SearchedLocations);
+            }
+
+            var view = viewEngineResult.EnsureSuccessful().View;
+
+            logger.LogVerbose("The partial view '{PartialViewName}' was found successfully.", viewName);
```

Comment:
Yeah it's an interesting discussion. I just wonder on a real site how many log entries this will generate. I guess here it's just in the view results so it's one per request. But should it be the view engine doing the logging instead?